src/libopenrave/kinbodylink.cpp

```diff
@@ -337,21 +340,33 @@ void KinBody::LinkInfo::DeserializeJSON(const rapidjson::Value &value, dReal fUn
         }
     }
 
-    _mapExtraGeometries.clear();
-    // if (value.HasMember("extraGeometries")) {
-    //     for (rapidjson::Value::ConstMemberIterator it = value["extraGeometries"].MemberBegin(); it != value["extraGeometries"].MemberEnd(); ++it) {
-    //         if (_mapExtraGeometries.find(it->name.GetString()) == _mapExtraGeometries.end()) {
-    //             _mapExtraGeometries[it->name.GetString()] = std::vector<GeometryInfoPtr>();
-    //         }
-    //         std::vector<GeometryInfoPtr>& vgeometries = _mapExtraGeometries[it->name.GetString()];
-    //         vgeometries.reserve(it->value.Size() + vgeometries.size());
-    //         size_t iGeometry = 0;
-    //         for(rapidjson::Value::ConstValueIterator im = it->value.Begin(); im != it->value.End(); ++im, ++iGeometry) {
-    //             std::string id = orjson::GetStringJsonValueByKey(*im, "id");
-    //             UpdateOrCreateInfoWithNameCheck(*im, id, vgeometries, "name", fUnitScale);
-    //         }
-    //     }
-    // }
+    if (value.HasMember("extraGeometries") && value["extraGeometries"].IsArray()) {
+        for (rapidjson::Value::ConstValueIterator it_l = value["extraGeometries"].Begin(); it_l != value["extraGeometries"].End(); ++it_l) {
```

Comment:
@zax147 Could you use a more relevant and meaningful iterator name other than `it_1`?

Reply:
I committed a new change to address this, and I keep the iterator name as `it` for the consistency with previous fields' iterators.